src/SmartComponents/ActivityTable/__tests__/ActivityTable.tests.js

```diff
@@ -215,8 +216,28 @@ describe('ActivityTable', () => {
 
     await waitFor(() => {
       expect(fetchExecutedTasks).toHaveBeenCalled();
-      userEvent.click(screen.getAllByLabelText('Kebab toggle')[0]);
-      userEvent.click(screen.getByText('Run this task again'));
+    });
+
+    const row = screen.getByRole('row', {
+      name: /task b 5 running 21 apr 2022, 10:10 utc/i,
+    });
+
+    await waitFor(() => {
+      //userEvent.click(screen.getAllByLabelText('Kebab toggle')[0]);
+      userEvent.click(
+        within(row).getByRole('button', {
+          name: /kebab toggle/i,
+        })
+      );
+    });
```

Comment:
This can be safely rewritten to 

```suggestion
    await userEvent.click(
        within(row).getByRole('button', {
          name: /kebab toggle/i,
        })
      );
```

According to the documentation, there is no need to use waitFor for userEvent methods.